server/router/server.go

```diff
@@ -48,170 +48,170 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int, g http.
+	gMux.Handle("/api/pattern/clone/{id}", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.CloneMesheryPatternHandler), models.ProviderAuth))).
 		Methods("POST")
-	gMux.Handle("/api/patterns/delete", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.DeleteMultiMesheryPatternsHandler)))).
+	gMux.Handle("/api/patterns/delete", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.DeleteMultiMesheryPatternsHandler), models.ProviderAuth))).
 		Methods("POST")
 
-	gMux.HandleFunc("/api/oam/{type}", h.OAMRegisterHandler).Methods("GET", "POST")
-	gMux.HandleFunc("/api/oam/{type}/{name}", h.OAMComponentDetailsHandler).Methods("GET")
-	gMux.HandleFunc("/api/oam/{type}/{name}/{id}", h.OAMComponentDetailByIDHandler).Methods("GET")
-	gMux.HandleFunc("/api/meshmodel/validate", h.ValidationHandler).Methods("POST")
-	gMux.HandleFunc("/api/meshmodel/component/generate", h.ComponentGenerationHandler).Methods("POST")
-	gMux.HandleFunc("/api/meshmodel/components", h.GetAllMeshmodelComponents).Methods("GET")
+	gMux.Handle("/api/oam/{type}", h.AuthMiddleware(http.HandlerFunc(h.OAMRegisterHandler), models.NoAuth)).Methods("GET", "POST")
+	gMux.Handle("/api/oam/{type}/{name}", h.AuthMiddleware(http.HandlerFunc(h.OAMComponentDetailsHandler), models.NoAuth)).Methods("GET")
+	gMux.Handle("/api/oam/{type}/{name}/{id}", h.AuthMiddleware(http.HandlerFunc(h.OAMComponentDetailByIDHandler), models.NoAuth)).Methods("GET")
+	gMux.Handle("/api/meshmodel/validate", h.AuthMiddleware(http.HandlerFunc(h.ValidationHandler), models.NoAuth)).Methods("POST")
+	gMux.Handle("/api/meshmodel/component/generate", h.AuthMiddleware(http.HandlerFunc(h.ComponentGenerationHandler), models.NoAuth)).Methods("POST")
+	gMux.Handle("/api/meshmodel/components", h.AuthMiddleware(http.HandlerFunc(h.GetAllMeshmodelComponents), models.NoAuth)).Methods("GET")
 
-	gMux.HandleFunc("/api/components", h.GetAllComponents).Methods("GET")
-	gMux.HandleFunc("/api/components/types", h.ComponentTypesHandler).Methods("GET")
-	gMux.HandleFunc("/api/components/{type}", h.ComponentsForTypeHandler).Methods("GET")
-	gMux.HandleFunc("/api/components/{type}/versions", h.ComponentVersionsHandler).Methods("GET")
-	gMux.HandleFunc("/api/components/{type}/{name}", h.ComponentsByNameHandler).Methods("GET")
+	gMux.Handle("/api/components", h.AuthMiddleware(http.HandlerFunc(h.GetAllComponents), models.NoAuth)).Methods("GET")
+	gMux.Handle("/api/components/types", h.AuthMiddleware(http.HandlerFunc(h.ComponentTypesHandler), models.NoAuth)).Methods("GET")
+	gMux.Handle("/api/components/{type}", h.AuthMiddleware(http.HandlerFunc(h.ComponentsForTypeHandler), models.NoAuth)).Methods("GET")
+	gMux.Handle("/api/components/{type}/versions", h.AuthMiddleware(http.HandlerFunc(h.ComponentVersionsHandler), models.NoAuth)).Methods("GET")
+	gMux.Handle("/api/components/{type}/{name}", h.AuthMiddleware(http.HandlerFunc(h.ComponentsByNameHandler), models.NoAuth)).Methods("GET")
```

Comment:
This collection of endpoints should be protected with ProviderAuth.

Reply:
Adapter will be registering MeshModel components on this endpoint, that is why they are left to NoAuth **for now**

Reply:
Any ways, these will be put under /meshmode/components

Reply:
@Revolyssup, only the POST endpoint needs NoAuth, though, right?